index.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs');
+const path = require('path');
+
+class AssetMap {
+
+  constructor(options) {
```

Comment:
lets also add an option for readOnLoad that will populate the cache on initialization